kitty/window.py

```diff
@@ -438,7 +438,6 @@ def destroy(self):
 
     def as_text(self, as_ansi=False, add_history=False, add_pager_history=False, add_wrap_markers=False, alternate_screen=False):
         lines = []
-        add_history = add_history and not self.screen.is_using_alternate_linebuf() and not alternate_screen
```

Comment:
The secondary screen (the one used by fullscreen applications) has no history buf. So if we are getting the text from that screen add_history must be false. which means that either the screen is not using alternate_linebuf and alternate screen is false or both are true. basically the condition should be

not (self.screen.is_using_alternate_linbebuf() ^ alternate_linebuf)

Reply:
Got it, thanks